### ISSUE TITLE
maybe a problem with SpcaingNormalizer?

### ISSUE THREAD

**Issue by sghamilouei** [amirivojdan/shekar#12]
```
from shekar.normalizer import SpacingNormalizer

text = """.این متن دارای نیم فاص‌له است"""

spn = SpacingNormalizer()
normalized_text = spn.fit_transform(text)
print(normalized_text)
```

yet the non joiner was still there.

thanks. this library is really helping me!

**Comment by amirivojdan** [comment]
Thank you very much for the detailed example and for taking the time to open this issue, I really appreciate the feedback.

Regarding the behavior you observed, the `SpacingNormalizer` is designed to fix known spacing patterns such as:
- common verbal spacings
- punctuation spacing
- standard compound words
- well-defined ZWNJ (half-space) conventions in Persian

However, in your example, **`فاص‌له`** is actually a typo (an incorrect internal split) rather than a standard spacing pattern. Because of that, it falls outside the scope of rule-based spacing normalization.

For cases like this, the spell checker is a more appropriate tool:

```python
from shekar import SpellChecker

spell_checker = SpellChecker()
print(spell_checker("فاص‌له"))
```

This approach targets lexical errors and misspellings, which is why it can better handle malformed words that the spacing normalizer intentionally does not modify.

Thanks again for reporting this, it helps improve the library and clarify the intended behavior of each component.
